ecs_k8s_ubuntu_16.04_master.sh

```diff
@@ -0,0 +1,76 @@
+# ------------------------------------------------------------------------------------------------------------------------
+# We are explicitly not using a templating language to inject the values as to encourage the user to limit their
+# use of templating logic in these files. By design all injected values should be able to be set at runtime,
+# and the shell script real work. If you need conditional logic, write it in bash or make another shell script.
+# ------------------------------------------------------------------------------------------------------------------------
+
+# Specify the Kubernetes version to use.
+KUBERNETES_VERSION="1.10.2"
+
+# Obtain IP addresses
+HOSTNAME=$(curl -s http://169.254.169.254/latest/meta-data/hostname | cut -d '.' -f 1)
+PUBLICIP=$(curl -s http://169.254.169.254/latest/meta-data/public-ipv4 )
+PRIVATEIP=$(ip -f inet -o addr show ens3|cut -d\  -f 7 | cut -d/ -f 1)
+
+# Add Kubernetes repository.
+curl -s https://packages.cloud.google.com/apt/doc/apt-key.gpg | sudo apt-key add -
+touch /etc/apt/sources.list.d/kubernetes.list
+sh -c 'echo "deb http://apt.kubernetes.io/ kubernetes-xenial main" > /etc/apt/sources.list.d/kubernetes.list'
+
+# Install packages.
+apt-get update -y
+apt-get install -y \
+    socat \
+    ebtables \
+    docker.io \
+    apt-transport-https \
+    kubelet=${KUBERNETES_VERSION}-00 \
+    kubeadm=${KUBERNETES_VERSION}-00 \
+    kubernetes-cni \
```

Comment:
To prevent installation failures when new Kubernetes and CNI versions are released, this should be bound to some version. You can use the latest version here, which is `0.6.0-00`